frontend_tests/node_tests/compose_ui.js

```diff
@@ -74,6 +74,58 @@ run_test('insert_syntax_and_focus', () => {
     assert.equal($('#compose-textarea').val(), 'xyz :octopus: ');
     assert($("#compose-textarea").is_focused());
 
+    // Tests for block_format option in various cases
+    var textarea = make_textbox('');
+    // Case 1: "|"
+    textarea.caret(0);
+    compose_ui.insert_syntax_and_focus(':sharky:', textarea, true);
+    assert.equal(textarea.val(), ':sharky:\n\n ');
```

Comment:
That extra space at the end was caused by a bug that I fixed in f40cbdbd1934a530d7768e9a19195153d6b94b32; can you rebase and fix the tests?